KubeArmor/core/kubeUpdate.go

```diff
@@ -2375,3 +2390,60 @@ func (dm *KubeArmorDaemon) WatchDefaultPosture() {
 		}
 	}
 }
+
+// WatchConfigMap function
+func (dm *KubeArmorDaemon) WatchConfigMap(namespace string, configMapName string) {
+	ns, err := K8s.K8sClient.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	if err != nil {
+		kg.Err("unable to fetch requested namespace")
+	}
+
+	cmWatcher, err := K8s.K8sClient.CoreV1().ConfigMaps(namespace).Watch(context.TODO(), metav1.SingleObject(metav1.ObjectMeta{
+		Name:      configMapName,
+		Namespace: namespace,
+	}))
+	defer cmWatcher.Stop()
+	if err != nil {
+		kg.Err("unable to create config map watcher")
+	}
+	for resp := range cmWatcher.ResultChan() {
+		if resp.Type == watch.Modified || resp.Type == watch.Added {
+			if cm, ok := resp.Object.(*corev1.ConfigMap); ok {
+				posture := tp.DefaultPosture{
+					FileAction:         validateDefaultPosture("kubearmor-file-posture", ns, cm.Data[cfg.ConfigDefaultFilePosture]),
+					NetworkAction:      validateDefaultPosture("kubearmor-network-posture", ns, cm.Data[cfg.ConfigDefaultNetworkPosture]),
+					CapabilitiesAction: validateDefaultPosture("kubearmor-capabilities-posture", ns, cm.Data[cfg.ConfigDefaultCapabilitiesPosture]),
+				}
+				cfg.GlobalCfg.HostVisibility = cm.Data[cfg.ConfigVisibility]
+				dm.UpdateGlobalPosture(posture)
+				// update default posture for endpoints
+
+				for _, endpoint := range dm.EndPoints {
+					ns := endpoint.NamespaceName
+					dm.Logger.Printf("updating default posture for %s in %s", ns, endpoint.EndPointName)
+					dm.UpdateDefaultPosture(string(resp.Type), ns, posture)
```

Comment:
If we have a namespace annotated, This would override that annotation tho?

Suppose we have namespace  "test" annotated with default posture block, and then we update config map to have audit posture. Namespace test should still be blocking.

Reply:
🤔  I could loop through dm.k8s pods and check annotations ?

Reply:
Ig maybe we should try some major refactor here? Like we don't fill up the posture map struct if they are not annotated. And if no annotation is found we check the global defaults.

If this is not feasible then yes Looping with specific conditions should do.